cmake/FindMetaExamples.cmake

```diff
@@ -15,40 +15,75 @@ function(is_set CONFIG_FILE FLAG)
 	endif()
 endfunction()
 
-function(find_meta_examples)
-	read_config_header(${shogun_INCLUDE_DIR}/shogun/lib/config.h)
-	is_set(${SHOGUN_CONFIG} HAVE_NLOPT)
-	is_set(${SHOGUN_CONFIG} USE_GPL_SHOGUN)
-	is_set(${SHOGUN_CONFIG} HAVE_LAPACK)
-	is_set(${SHOGUN_CONFIG} USE_SVMLIGHT)
-
-	FILE(GLOB_RECURSE META_EXAMPLE_LISTINGS ${CMAKE_SOURCE_DIR}/examples/meta/src/*.sg)
+function(get_excluded_meta_examples)
+    read_config_header(${shogun_INCLUDE_DIR}/shogun/lib/config.h)
+    is_set(${SHOGUN_CONFIG} HAVE_NLOPT)
+    is_set(${SHOGUN_CONFIG} USE_GPL_SHOGUN)
+    is_set(${SHOGUN_CONFIG} HAVE_LAPACK)
+    is_set(${SHOGUN_CONFIG} USE_SVMLIGHT)
 
-	# temporary hacks to exclude certain meta examples that have dependencies
-	IF(NOT HAVE_NLOPT)
-		LIST(REMOVE_ITEM META_EXAMPLE_LISTINGS
-			${CMAKE_SOURCE_DIR}/examples/meta/src/gaussian_processes/gaussian_process_regression.sg)
+    IF(NOT HAVE_NLOPT)
+		LIST(APPEND EXCLUDED_META_EXAMPLES
+			gaussian_processes/gaussian_process_regression.sg)
 	ENDIF()
 
 	IF(NOT USE_GPL_SHOGUN)
-		LIST(REMOVE_ITEM META_EXAMPLE_LISTINGS
-			${CMAKE_SOURCE_DIR}/examples/meta/src/gaussian_processes/gaussian_process_regression.sg
-			${CMAKE_SOURCE_DIR}/examples/meta/src/multiclass_classifier/multiclass_logisticregression.sg
+		LIST(APPEND EXCLUDED_META_EXAMPLES
+			gaussian_processes/gaussian_process_regression.sg
+			multiclass_classifier/multiclass_logisticregression.sg
 			)
 	ENDIF()
 
 	IF(NOT HAVE_LAPACK)
-		LIST(REMOVE_ITEM META_EXAMPLE_LISTINGS
-			${CMAKE_SOURCE_DIR}/examples/meta/src/regression/linear_ridge_regression.sg
-			${CMAKE_SOURCE_DIR}/examples/meta/src/clustering/gmm.sg
-			${CMAKE_SOURCE_DIR}/examples/meta/src/distance/mahalanobis.sg
+		LIST(APPEND EXCLUDED_META_EXAMPLES
+			regression/linear_ridge_regression.sg
+			clustering/gmm.sg
+			distance/mahalanobis.sg
 			)
 	ENDIF()
 
 	IF(NOT USE_SVMLIGHT)
-		LIST(REMOVE_ITEM META_EXAMPLE_LISTINGS
-			${CMAKE_SOURCE_DIR}/examples/meta/src/regression/multiple_kernel_learning.sg)
+		LIST(APPEND EXCLUDED_META_EXAMPLES
+			regression/multiple_kernel_learning.sg)
 	ENDIF()
 
+    SET(EXCLUDED_META_EXAMPLES ${EXCLUDED_META_EXAMPLES} PARENT_SCOPE)
+
+endfunction()
+
+# Remove meta example that cannot be built because of missing dependencies
+function(find_meta_examples)
+
+	FILE(GLOB_RECURSE META_EXAMPLE_LISTINGS ${CMAKE_SOURCE_DIR}/examples/meta/src/*.sg)
+    get_excluded_meta_examples()
+
+	# temporary hacks to exclude certain meta examples that have dependencies
+    FOREACH(META_EXAMPLE ${EXCLUDED_META_EXAMPLES})
+        LIST(REMOVE_ITEM META_EXAMPLE_LISTINGS ${CMAKE_SOURCE_DIR}/examples/meta/src/${META_EXAMPLE})
+    ENDFOREACH()
+
 	SET(META_EXAMPLES ${META_EXAMPLE_LISTINGS} PARENT_SCOPE)
 endfunction()
+
+# Get the cookbook pages we want to exclude from build
+function(find_excluded_meta_examples)
+
+    get_excluded_meta_examples()
+
+    FOREACH(META_EXAMPLE ${EXCLUDED_META_EXAMPLES})
+        LIST(APPEND EXCLUDED_COOKBOOK_PAGES examples/${META_EXAMPLE})
```

Comment:
one thing to consider here: Some meta examples do not have cookbook pages. So they do not need to be excluded technically. Not sure whether that causes problems, but it is definitely cleaner to only work on the existing cookbook pages